### ISSUE TITLE
v1.71 and 1.80 SetInstance breaks object creation

### ISSUE THREAD

**Issue by kszepieto** [darrencauthon/AutoMoq#29]
Below test fails on v.1.80 and 1.71 but passes on v1.70
```
namespace AutoMoqIssue
{
    [TestClass]
    public class AutoMoqTests
    {
        [TestMethod]
        public void can_create_parent_object_when_setInstance_is_called_on_child()
        {
            var autoMoq = new AutoMoqer();
            
            var child = autoMoq.Create<Child>();
            autoMoq.SetInstance<IChild>(child);

            var parent = autoMoq.Create<Parent>();
            Assert.IsNotNull(parent);
        }
    }

    public interface IParent
    {
    }

    public interface IChild
    {
    }

    public interface IGrandChild
    {
    }
    
    public class Parent : IParent
    {
        private readonly IChild _child;
        private readonly IGrandChild _grandChild;

        public Parent(IChild child, IGrandChild grandChild)
        {
            _child = child;
            _grandChild = grandChild;
        }
    }

    public class Child : IChild
    {
        private readonly IGrandChild _grandChild;

        public Child(IGrandChild grandChild)
        {
            _grandChild = grandChild;
        }
    }

    public class GrandChild : IGrandChild
    {
    }
}
```

**Comment by darrencauthon** [comment]
Thank you for the failing test.  I've started a pull request to resolve it.  I'll post an update as soon as I have a fix.

**Comment by darrencauthon** [comment]
@kszepieto The issue seems to be calling Create twice, so I believe that's something I'll be able to resolve.

I haven't seen this in any test because I've never come across that... calling ```Create<T>``` twice.  In my thousands of tests written with this tool, I've always had only one call to that method per test.  But it should work!  This test of yours will be great to prevent the regression.

If you are stuck in a bad spot right now, you can downgrade to 1.7.1 or use Moq to create the stub object.  I do this occassionally:

```
var foo = new Mock<IFoo>().Object;
```

**Comment by darrencauthon** [comment]
@kszepieto I'm still investigating.

It is a matter of calling ```Resolve``` twice during a test -- not the ```SetInstance``` method.  If you use ```Resolve``` only to register the class under test, you should be fine.

**Comment by darrencauthon** [comment]
I believe I resolved the issue here:  

https://github.com/darrencauthon/AutoMoq/pull/30/commits/e7d3977774c1ea47f90c5eb9e20f7a0ac6eb6245

That fixed the new failing tests.  I'm still getting an error around abstract dependency resolution, which I'm still in the process of resolving. 

The ultimate fail-safe is to remove the refactorings I made to isolate the "mocking" and "ioc" work.  But I really don't want to give them up quite yet -- they're essential to the next place I want to take AutoMoq.  Still working...

**Comment by darrencauthon** [comment]
The fix for this has been published as 1.8.1.

Thanks @kszepieto for the catch!  It's strange, in all of the years of using this library I've never called ```Create<T>``` twice in one test!  Your unit tests are part of the test suite to prevent regressions.
